Sources/PaymentsCore/Networking/HTTP.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// `HTTP` interfaces directly with `URLSession` to execute network requests.
+class HTTP {
+    
+    private let coreConfig: CoreConfig
+    private var urlSession: URLSessionProtocol
+    private let decoder = APIClientDecoder()
+
+    init(coreConfig: CoreConfig) {
+        self.urlSession = URLSession.shared
+        self.coreConfig = coreConfig
+    }
+
+    /// For internal use for testing purpose
+    init(urlSession: URLSessionProtocol, coreConfig: CoreConfig) {
+        self.urlSession = urlSession
+        self.coreConfig = coreConfig
+    }
+    
+    func performRequest<T: APIRequest>(endpoint: T) async throws -> (T.ResponseType) {
+        guard let request = endpoint.toURLRequest(environment: coreConfig.environment) else {
+            throw APIClientError.invalidURLRequestError
+        }
+        let (data, response) = try await urlSession.performRequest(with: request)
+        guard let response = response as? HTTPURLResponse else {
+            throw APIClientError.invalidURLResponseError
+        }
+
+        switch response.statusCode {
```

Comment:
Should we add a todo to handle the other status codes here? Or do we just want to handle 200-300 the same and everything else as a server response? I think eventually we'd want to handle 400 and 500 level errors differently then have a default for everything else. Thoughts?

Reply:
Good question!

[It looks like on Android](https://github.com/paypal/Android-SDK/blob/e8f8aa29e07f1bf5600f8080c7d4046c69b825a0/Core/src/main/java/com/paypal/android/core/HttpResponse.kt#L16), we have the same check (200 to 300) is a success

It might be nice though to have more granular error handling per code, at least 400 v 500. I think this could be a follow-up PR / investigation to see if we think this would add value to our merchants / make our lives easier when troubleshooting. What do you think?

Reply:
Yeah, I think it makes sense to be a follow up/alignment on both platforms. Agree that it likely would make merchants and our troubleshooting easier.